### PR TITLE
Additional conditions

### PR DESCRIPTION
**Pull request by Vierdant**
### **Desc:**
Added new conditions that I personally would use.

## **HasPermission**
Checks if the target (player) has a permission node

Example: `?HasPermission{permission=essentials.god}`

## **OnGround**
A simple condition that checks if the target is standing on solid ground, or a solid specified block.

Example: `?OnGround{}`

On a specific block(s) example: `?OnGround{blocks=DIRT,STONE,GRAVEL}`

## **MythicMobsFaction** - **Requires mythic mobs**
Checks if the target entity is of a provided faction

Example: `?MythicMobsFaction{faction=monsters}`

### CODE REVIEW

File: MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/HasPermissionCondition.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    public HasPermissionCondition(String permission) {
+        this.permission = permission;
+    }
+
+    @Override
+    protected boolean isAllowed0(CastData cast) {
+        LivingEntity target = cast.getTarget();
+        if (target == null) return false;
+        if (!(target instanceof Player)) return false;
```
Comment (CJCrafter) [inline MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/HasPermissionCondition.java:25]: This check should be removed

File: MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/MythicMobsFactionCondition.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public MythicMobsFactionCondition(String faction) {
+        this.faction = faction;
+    }
+
+    @Override
+    protected boolean isAllowed0(CastData cast) {
+        LivingEntity target = cast.getTarget();
+        if (target == null) return false;
+
+        if (target instanceof Player) {
```
Comment (CJCrafter) [inline MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/MythicMobsFactionCondition.java:32]: You should be using the [enhanced instanceof](https://www.baeldung.com/java-pattern-matching-instanceof#pattern-matching) from Java 14

File: MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/OnGroundCondition.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package me.vierdant.bridgedmechanics.tmep;
+
+import me.deecaad.core.file.SerializeData;
+import me.deecaad.core.file.SerializerException;
+import me.deecaad.core.mechanics.CastData;
+import me.deecaad.core.mechanics.conditions.Condition;
+import org.jetbrains.annotations.NotNull;
+
+public class OnGroundCondition extends Condition {
```
Comment (CJCrafter) [inline MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/OnGroundCondition.java:9]: `OnBlock` and `OnGround` need to be merged... less clutter on the wiki.

Imagine
```yaml
Mechanics:
  - "?OnGround{}"  # Checks if the entity is on the ground
  - "?OnGround{blocks=[DIRT, GRAVEL]}"  # Checks if the entity is on the ground AND that block is dirt or gravel. 
```

File: MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/OutsideCondition.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package me.deecaad.core.mechanics.conditions;
+
+import me.deecaad.core.file.SerializeData;
+import me.deecaad.core.file.SerializerException;
+import me.deecaad.core.mechanics.CastData;
+import me.deecaad.core.mechanics.conditions.Condition;
+import org.bukkit.Location;
+import org.jetbrains.annotations.NotNull;
+
+public class OutsideCondition extends Condition {
```
Comment (CJCrafter) [inline MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/OutsideCondition.java:10]: Have to reject this one... This is already done and more efficient by using the [LightLevelCondition](https://github.com/WeaponMechanics/MechanicsMain/wiki/LightLevelCondition) and making sure it is 16 (maximum light)

File: MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/MythicMobsFactionCondition.java (hunk 2 above)
Comment (CJCrafter) [inline MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/MythicMobsFactionCondition.java:32]: You should be using the [enhanced instanceof](https://www.baeldung.com/java-pattern-matching-instanceof#pattern-matching) from Java 14

**Review by CJCrafter — CHANGES_REQUESTED** [review]
Thanks for contributing! Your changes look pretty good, but I left a few comments before I can merge them.

**Comment by Vierdant** [comment]
Requested changes implemented! Removed unnecessary conditions and improved the code.

**Review by CJCrafter — APPROVED** [review]
Looks good! I'll make a wiki for each page then get this merged. Thanks for contributing

**Pull request merged**